### PR TITLE
Huobi: Add subscription configuration

### PR DESCRIPTION
**Pull request by gbjk**
Just adds configurable subs.

This PR does not fix auth WS usage, which appears broken due to using deprecated v1 paths.
Also doesn't seek to make subs synchronous.

## Stacked Dependencies

- [x] #1579 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: exchanges/huobi/huobi_websocket.go
Comment (shazbert) [inline exchanges/huobi/huobi_websocket.go:66]: Suggestion: Levels: 0 looks a bit weird, any chance we can get some commentary on that? Or package const for blank values? 

NOTE: Also orderbook levels don't come up in the config so it's not as obvious that you can configure it there.

**Review by shazbert — CHANGES_REQUESTED** [review]
Good stuff, minor potential nit

File: exchanges/huobi/huobi_websocket.go
Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go:66]: Propose we drop Levels 0 from this, and add docs to the doc template.

Comment (shazbert) [reply exchanges/huobi/huobi_websocket.go:66]: Sounds good. 👍

Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go]: Fixed aef34014 and 554f69c9

**Review by shazbert — APPROVED** [review]
LGTM.

File: exchanges/subscription/list.go
Comment (gloriousCode) [inline exchanges/subscription/list.go:59]: Match the func elsewhere, or wait until the other one is merged 🤙

File: exchanges/huobi/huobi_websocket.go
Comment (gloriousCode) [inline exchanges/huobi/huobi_websocket.go:66]: Cannot be customisable to the user?
Your readme says:
>- Configure Level: 1-5 for depth aggregation

Config:
```
  {
      "enabled": true,
      "channel": "orderbook",
      "asset": "spot"
     },
```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
>Also doesn't seek to make subs synchronous

Why not? Huobi has id support which would allow for SendMessageReceiveResponse

**Comment by gbjk** [comment]
> > Also doesn't seek to make subs synchronous
> 
> Why not? Huobi has id support which would allow for SendMessageReceiveResponse

It's just not the focus of this PR. I have a list of things to fix on Huobi after this is merged, and synchronous `Subscribe` is one of them.
The PRs that touch everything at once are painful to review, painful to merge, and conflict often.
So I'm trying to work with more agility, limited blast zones, and iterative improvements. 🙂 

That said: I can sniff making it synchronous now it it's a dealbreaker for this to be merged.

File: exchanges/subscription/list.go
Comment (gbjk) [reply exchanges/subscription/list.go]: Fixed aa10abb3c

File: exchanges/huobi/huobi_websocket.go
Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go:66]: Did you see the line above in the README?
```
Default Public Subscriptions:
- Ticker
- Candles ( Interval: 1min )
- Orderbook ( Level: 0 - No aggregation )
  - Configure Level: 1-5 for depth aggregation
- Trades
```

Since level 0 is "don't aggregate", which is basically normal, it feels okay that it's omitted from the config.

Fixed d7896584d

Is this acceptable?

Comment (gloriousCode) [reply exchanges/huobi/huobi_websocket.go:66]: I did see the level, but I wanted a JSON example to make it clear what's required for a user. Thanks for the update

File: exchanges/huobi/huobi_test.go
Comment (gloriousCode) [inline exchanges/huobi/huobi_test.go:3060]:
```
    huobi_test.go:3060: Websocket not enabled
--- SKIP: TestSubscribe (0.01s)
```
🤷 There seems to be no Github Actions running on this pr too

Anyway, I request a mock subscribe test here too.

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Tested OB levels ✅ 
One more thing

File: exchanges/huobi/huobi_test.go
Comment (gbjk) [reply exchanges/huobi/huobi_test.go:2975]: I implemented mocks for authed subs, but it also meant implementing V2 auth, and the rabbit hole became entirely too much.
So I've enabled websocket and commented out auth endpoints in the translation table.

The [v2 implementation](https://github.com/gbjk/gocryptotrader/pull/new/feature/huobi_v2) will be ready very soon, and that commenting can go away.

I've deliberately left the authed subs in config so that we don't need to do a config upgrade to enable v2.
Anyone using authed WS is broken already and won't be any more broken as a result of merging this PR.

Fixed 86b8b4c0

**Comment by gbjk** [comment]
> That said: I can sniff making it synchronous now it it's a dealbreaker for this to be merged.

This became too much for this PR. It just snowballed endlessly. I'm nearly done, but it should not get stacked on top of this.

So let's review this in the context of the title only.

File: exchanges/huobi/huobi_test.go
Comment (gloriousCode) [reply exchanges/huobi/huobi_test.go:2978]: Well I appreciate that v2 WS is on its way 🎉 
With all of that and no auth in this PR, I will delay demanding auth sub coverage to that future PR.

File: exchanges/huobi/huobi_websocket.go
Comment (gloriousCode) [inline exchanges/huobi/huobi_websocket.go:814]: I'm fine with the error text, but `ErrPrivateChannelName` implies that it could only be an auth string that gets here, even with the Authenticated check above. It could be any garbage channel here eg `lkasjhdfalk` in my config, not necessarily private.
```[ERROR] | WEBSOCKET | 05/11/2024 10:19:29 | Huobi websocket: subscription failure, LOLBUTTS spot : template: master.tmpl:5:7: executing "master.tmpl" at <channelName $.S $p>: error calling channelName: must use standard channel name constants```

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for making those changes, one petty request

File: exchanges/huobi/huobi_websocket.go
Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go]: Fixed e9f67fe99

**Review by gloriousCode — APPROVED** [review]
tACK

File: exchanges/huobi/huobi_test.go
Comment (thrasher-) [inline exchanges/huobi/huobi_test.go:2982]:
```suggestion
	require.Equal(t, 4, len(got), "Must get correct number of subscriptions")
```

**Review by thrasher- — CHANGES_REQUESTED** [review]
All looks good, only minor things with the known caveat of disabled auth subs

File: exchanges/huobi/huobi_websocket.go
Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go:584]: Fixed 94b3c7ec4

File: exchanges/huobi/huobi_websocket.go
Comment (gbjk) [reply exchanges/huobi/huobi_websocket.go:558]: Fixed 8cbed5b9e

File: exchanges/huobi/huobi_test.go
Comment (gbjk) [reply exchanges/huobi/huobi_test.go]: Fixed 8bbe1919f

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes!

**Review by shazbert — APPROVED** [review]
Good stuff tACK!

**Pull request merged**
